src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V66/Eth66ProtocolHandler.cs

```diff
@@ -276,12 +278,51 @@ protected override async Task<TxReceipt[][]> SendRequest(V63.Messages.GetReceipt
             }
 
             GetReceiptsMessage msg66 = new() { EthMessage = message };
-            return await SendRequestGeneric(
+            return await SendRequestGenericEth66(
                 _receiptsRequests66,
                 msg66,
                 TransferSpeedType.Receipts,
                 static (_) => $"{nameof(GetReceiptsMessage)}",
                 token);
         }
+
+        private async Task<TResponse> SendRequestGenericEth66<T66, TRequest, TResponse>(
+            MessageDictionary<T66, TRequest, TResponse> messageQueue,
+            T66 message,
+            TransferSpeedType speedType,
+            Func<T66, string> describeRequestFunc,
+            CancellationToken token
+        )
+            where T66 : Eth66Message<TRequest>
+            where TRequest : P2PMessage
+        {
+            message.RequestId = Interlocked.Increment(ref _currentRequestId);
+            Request<T66, TResponse> request = new(message);
+            messageQueue.Send(request);
```

Comment:
all code under this line is same as SendRequestGeneric, maybe should be in some reused method?

Reply:
Done